From 3e5e7fd6c93517278b2451a08f47e16d052babbb Mon Sep 17 00:00:00 2001
From: David Teigland <teigland@redhat.com>
Date: Tue, 10 Sep 2019 09:47:33 -0500
Subject: [PATCH] pvscan: allow use of noudevsync option

When pvscan is used to activate a VG via an
asynchronous service (i.e. lvm2-pvscan), there
is no requirement that the command wait for
udev to create device nodes before returning.

It's possible that waiting for udev is slow
enough to cause the service running the command
to time out.  So, allow the --noudevsync option
to be given to pvscan to skip waiting for udev.

(This commit is not changing the lvm2-pvscan
service itself to use --noudevsync.)

Still unknown is whether there are any complex
LV activation cases in which lvm itself requires
access to a device node, in which case the udev
wait could be needed by lvm itself.

(When running an activation command directly
from the command line, it's generally expected
that the activated LVs are ready to use when
the command is finished, so lvm waits for
udev to finish creating the dev nodes.)
---
 tools/command-lines.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/command-lines.in b/tools/command-lines.in
index 1169224792..65221c0d1a 100644
--- a/tools/command-lines.in
+++ b/tools/command-lines.in
@@ -1498,7 +1498,7 @@ DESC: Display PV information.
 
 pvscan --cache_long
 OO: --ignorelockingfailure, --reportformat ReportFmt,
---activate ay, --major Number, --minor Number
+--activate ay, --major Number, --minor Number, --noudevsync
 OP: PV|String ...
 IO: --background
 ID: pvscan_cache
-- 
2.21.0

